### PR TITLE
Stop server support

### PR DESCRIPTION
**Pull request by sriramgd**
From cubitouch's gist. Tested on ubuntu and windows from omnisharp-vim.

### CODE REVIEW

File: src/OmniSharp/Api/Server/OmnisharpController.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        protected readonly IApplicationShutdown _applicationShutdown;
+        public OmnisharpController(IApplicationShutdown applicationShutdown)
+        {
+            _applicationShutdown = applicationShutdown;
+        }
+
+        [HttpPost("/stopserver")]
+        public void StopServer()
+        {
+            _applicationShutdown.RequestShutdown();
```
Comment (davidfowl) [inline src/OmniSharp/Api/Server/OmnisharpController.cs]: Do a Task.Run here:

```C#
Task.Run( _applicationShutdown.RequestShutdown);
```

File: src/OmniSharp/Api/Server/OmnisharpController.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.AspNet.Mvc;
+using Microsoft.Framework.Runtime;
+
+namespace OmniSharp.Api.Server
+{
+    public partial class OmnisharpController : Controller
+    {
+        protected readonly IApplicationShutdown _applicationShutdown;
```
Comment (nosami) [inline src/OmniSharp/Api/Server/OmnisharpController.cs]: this should be private

File: src/OmniSharp/Api/Server/OmnisharpController.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.AspNet.Mvc;
+using Microsoft.Framework.Runtime;
+
+namespace OmniSharp.Api.Server
+{
+    public partial class OmnisharpController : Controller
```
Comment (nosami) [inline src/OmniSharp/Api/Server/OmnisharpController.cs]: This should be a new controller as it has different dependencies to OmniSharpController. Should be
```public class StopServerController```

**Comment by sriramgd** [comment]
Hey guys, Could you decline this request. After some more testing, I notice that kestrel throws a 'EBUSY resource busy or locked' sometimes on ubuntu. The server stops, but the exception is unhandled.

**Comment by nosami** [comment]
Sure. Thanks anyway :)

**Pull request closed without merging**